crates/model/src/order.rs

```diff
@@ -614,6 +614,8 @@ pub struct OrderMetadata {
     /// Does not take partial fill into account.
     #[serde(default, with = "u256_decimal")]
     pub solver_fee: U256,
+    #[serde(default, with = "u256_decimal")]
+    pub executed_solver_fee: U256,
```

Comment:
This is a bit unfortunate. I'm not sure this needs to be added to the `OrderMetadata` as it seems like a value that is only used by us internally. It's currently required because `on_settlement_event_updater.rs` uses the `model::Order` [[0](https://github.com/cowprotocol/services/blob/main/crates/autopilot/src/database/orders.rs#L173)] instead of the `database::orders::FullOrder` or some new type that contains exclusively the data needed for the surplus fee computation.
Improving the separation of concerns by not relying on `database::orders::FullOrder` would not leak this implementation detail into the `OrderMetadata`.

Reply:
This is a general problem with us re-using the same type in for the api and internally. You want to use `orders_in_tx` and for that you need to add this field. Doing it differently feels like to big of a change for this PR so I would keep it as is right now, but add a comment to this new field that it is only for internal use.

In the future we definitely should separate the API order type from internal order types. Then we won't have problems like this.

Reply:
Makes sense. But then we should at least not expose the field publicly by adding `#[serde(skip_serializing)]` to it.

Reply:
I will leave it as is (but under Option<>) for now. But will probably directly use `FullOrder` when I do refactor described here https://github.com/cowprotocol/services/issues/1286

Reply:
> Doing it differently feels like to big of a change for this PR

I don't think so. We just need to have custom order loading code just for `on_settlement_event_updater` module (it doesn't affect any other existing interface).